code/weapon/weapons.cpp

```diff
@@ -7497,7 +7524,55 @@ void weapon_hit( object * weapon_obj, object * other_obj, vec3d * hitpos, int qu
 		weapon_hit_do_sound(other_obj, wip, hitpos, armed_weapon, quadrant);
 	}
 
-	if (wip->impact_weapon_expl_effect.isValid() && armed_weapon) {
+
+ 	bool valid_conditional_impact = false;
+	int relevant_armor_idx = -1;
+	float relevant_fraction = 1.0f;
+	float hit_angle = 0.0f;
+	vec3d reverse_incoming = (vmd_zero_vector - weapon_obj->orient.vec.fvec);
+
+	if (hitnormal) {
+		hit_angle = vm_vec_delta_ang(hitnormal, &reverse_incoming, NULL);
+	}
+
+	if (other_obj != nullptr && other_obj->type == OBJ_SHIP) {
+		shipp = &Ships[other_obj->instance];
+		if (quadrant == -1) {
+			relevant_armor_idx = shipp->armor_type_idx;
+			relevant_fraction = other_obj->hull_strength / shipp->ship_max_hull_strength;
+		} else {
+			relevant_armor_idx = shipp->shield_armor_type_idx;
+			relevant_fraction = ship_quadrant_shield_strength(other_obj, quadrant);
+		}
+		if (wip->conditional_impacts.count(relevant_armor_idx) == 1) {
+			SCP_vector<ConditionalImpact> cis = wip->conditional_impacts[relevant_armor_idx];
```

Comment:
You could make this a `const &auto` reference, which avoids copying the vector over, and makes it a little less busy.  `const &SCP_vecor<ConditionalImpact>` is acceptable as well.

Reply:
Edit, I'm tired and I think it's `const auto&` and `const &SCP_vector<CondintionalImpact>